streamly-process.cabal

```diff
@@ -50,6 +50,15 @@ flag use-native
 
 common compile-options
   default-language: Haskell2010
+  if os(darwin)
+    cpp-options:    -DCABAL_OS_DARWIN
+
+  if os(linux)
+    cpp-options:    -DCABAL_OS_LINUX
+
+  if os(windows)
+    cpp-options:    -DCABAL_OS_WINDOWS
```

Comment:
Should not have to define our own host macros via cabal and pass them when GHC anyway supplies such macros.

For windows  `mingw32_HOST_OS`  is always defined by GHC.